pkg/scheduler/scheduler.go

```diff
@@ -263,11 +263,17 @@ func findFlavorForResource(
 	name corev1.ResourceName,
 	val int64,
 	wUsed map[string]int64,
+	resourceFlavors map[string]*kueue.ResourceFlavor,
 	cq *cache.ClusterQueue,
 	spec *corev1.PodSpec) (string, int64) {
 	// We will only check against the flavors' labels for the resource.
 	selector := flavorSelector(spec, cq.LabelKeys[name])
-	for _, flavor := range cq.RequestableResources[name] {
+	for _, flvLimit := range cq.RequestableResources[name] {
+		flavor, exist := resourceFlavors[flvLimit.Name]
+		if !exist {
+			log.Error(nil, "Flavor %v not found", flvLimit.Name)
```

Comment:
> Are you assuming that #134 would block the workloads at a different point and, thus, reaching this line will be an error?

Yes, but even if we don't have that, shouldn't we have an error in the logs on missing flavors? 

> In the meantime, I think we should return "", 0 here.

This will be strange if the CQ was part of a cohort, workloads from other CQ will be able to borrow the resources of this CQ and start, which this CQ's workloads will not be able to.